backend/AAiT-backend-group-12/domain/domain.go

```diff
@@ -0,0 +1,40 @@
+package domain
+
+import (
+	"time"
+)
+
+type Blog struct {
+	ID         string
+	Title      string
+	Content    string
+	UserID     string
+	Tags       []string
+	CreatedAt  time.Time
+	UpdatedAt  time.Time
+	ViewCount  uint
+	LikedBy    []User
+	DislikedBy []User
+	Comments   []Comment
+}
+
+// User represents a user entity in the domain.
+type User struct {
+	ID        string
+	UserName  string
+	Email     string
+	Password  string
+	Role      string
+	CreatedAt time.Time
+}
+
+// Comment represents a comment entity in the domain.
+type Comment struct {
+	ID        string
+	Content   string
+	UserID    string
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	ViewCount uint
+	Comments  []Comment
```

Comment:
I don't understand why we need the comments field on the comment struct.

Reply:
What if there is a reply for the comment also that is what we agreed on.